src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1541,35 +1541,53 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
             raise CLIError("Location is invalid. Use: az functionapp list-consumption-locations")
         functionapp_def.location = consumption_plan_location
         functionapp_def.kind = 'functionapp'
+        if is_linux and not runtime:
+            raise CLIError("usage error: --runtime required. Supported values 'dontnet', 'node', 'python'.")
```

Comment:
>dontnet [](start = 79, length = 7)

dotnet ;)

Reply:
@yugangw-msft  regarding --linux-mode i changed it to --function-app-mode to clarify this is the mode for the function app & is currently supported for linux apps only & once we have the backend support for windows as well, will update the code for this check out of is--linux